src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -162,16 +167,26 @@ private static void analyzeRepos(String outputPath, List<RepoConfiguration> conf
             }
 
             try {
-                GitCheckout.checkout(config.getRepoRoot(), config.getBranch());
+                GitRevParse.assertBranchExists(config, FileUtil.getBareRepoPath(config));
+                GitLsTree.validateFilePaths(config, FileUtil.getBareRepoPath(config));
+            } catch (GitBranchException gbe) {
+                logger.log(Level.SEVERE, String.format(MESSAGE_BRANCH_DOES_NOT_EXIST,
+                        config.getBranch(), config.getLocation()), gbe);
+                generateEmptyRepoReport(repoReportDirectory.toString(), Author.NAME_FAILED_TO_CLONE_OR_CHECKOUT);
+                continue;
+            } catch (InvalidFilePathException ipe) {
+                generateEmptyRepoReport(repoReportDirectory.toString(), Author.NAME_FAILED_TO_CLONE_OR_CHECKOUT);
+                continue;
+            }
```

Comment:
The exception is getting messy. We may need to improve it in another PR.

Reply:
Yes, there shouldn't be 2 try catch within a method. It is an indication of insufficient SLAP here.

Reply:
I don't think it would be a good idea to just combine all try catch blocks into one as well as it may be harder to debug if things fail.

Reply:
>I don't think it would be a good idea to just combine all try catch blocks into one as well as it may be harder to debug if things fail.

Which is why you will need to increase the level of abstraction, combining some of the codes and handle some of the raw exceptions at lower level.
Ideally if the exceptions are unique to each individual line of code, there wouldn't be problem combining all the try catch blocks into it.

Reply:
Noted